### PR TITLE
✨ Add Data Mapper (#49)

### PR DESCRIPTION
**Pull request by mistryrn**
* Add initial version of Data Mapper service to `apps/` dir
* Update root `package.json` with `data-mapper-dev` command to run the Data Mapper

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
   "scripts": {
     "build": "pnpm -r run build",
     "das-dev": "pnpm --filter '*-das' run dev",
+    "data-mapper-dev": "pnpm --filter data-mapper run dev",
```
Comment (anncatton) [inline package.json:9]: 👍

**Review by anncatton — APPROVED** [review]
I have nothing more insightful to add than "this is awesome" 😛 Thanks for all the thorough work @mistryrn !

File: apps/data-mapper/src/routers/participants.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+// get participant by id
+router.get('/:participantId', async (req, res) => {
+	logger.info('GET /participant/:participantId');
+	const { participantId } = req.params;
+	try {
+		const participant = await getParticipant(participantId);
+		res.status(200).send({ participant });
+	} catch (error) {
+		logger.error(error);
+		res.status(404).send({ error: 'Participant not found' });
```
Comment (justincorrigible) [inline apps/data-mapper/src/routers/participants.ts:38]: may I start a 204 vs 400 vs 404 debate? (jk)

File: apps/data-mapper/src/service/create.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+// KEYS-DAS
+// TODO: add Type instead of any
+const createParticipantOhipKey = async (participantId: string): Promise<any> => {
+	const { keysDasUrl } = getAppConfig();
+	// TODO: use urlJoin
+	// TODO: add error handling
+	const result = await fetch(`${keysDasUrl}/ohip-keys`, {
+		method: 'POST',
+		body: JSON.stringify({ participantId }),
+		headers: { 'Content-Type': 'application/json' },
```
Comment (justincorrigible) [inline apps/data-mapper/src/service/create.ts:68]: using axios, these headers can be abstracted to an axios instance. DRY

File: apps/data-mapper/src/service/search.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
+// TODO: fix urlJoin error
+// import urlJoin from 'url-join';
+
+import { getAppConfig } from '../config';
+
+// combines data from respective DASes to return a single Participant object
+// TODO: add Type instead of any
+export const getParticipant = async (participantId: string): Promise<any> => {
```
Comment (justincorrigible) [inline apps/data-mapper/src/service/search.ts:8]: I'd advocate for moving this function's definition to the end of the file, so it doesn't invoke the other functions before they are defined (lines 9-12).

While hoisting may make this moot, code is more readable when variables are defined before they are used.

File: apps/data-mapper/src/service/create.ts (hunk 3 above)
Comment (mistryrn) [reply apps/data-mapper/src/service/create.ts:68]: Great idea, thank you! I believe we are transitioning to using `axios` for HTTP requests going forward in this project (Alyssa and Sam are starting to use it in our `consent-ui` workspace), so I'll add some additional `TODO`s and create a ticket to track the work for switching from `fetch` to `axios` :)

File: apps/data-mapper/src/service/search.ts (hunk 4 above)
Comment (mistryrn) [reply apps/data-mapper/src/service/search.ts:8]: Another good catch! Will update shortly 🙇

File: apps/data-mapper/src/service/create.ts (hunk 3 above)
Comment (mistryrn) [reply apps/data-mapper/src/service/create.ts:68]: Ticket created to track fetch -> axios transition work here: https://github.com/OHCRN/platform/issues/72

**Pull request merged**